### ISSUE TITLE
Waiting for a Request is Broken if the Hash Changes

### ISSUE THREAD

**Issue by urbicon** [cypress-io/cypress#5629]
### Current behavior:
When a XHR request is followed by a change of the hash without changing the location, waiting for that request is impossible. In Single Page Applications, it is common practice to connect the navigation to the hash, but with the current behavior, we cannot wait for a XHR request that gets triggered by navigating on the page. 
 
```javascript
cy.server()
cy.route('**/api').as('api')
<request gets triggered, followed by a change of the hash>
cy.wait('@api')
```

This will produce a timeout.

### Desired behavior:

When the hash in the url is changed, it should not affect the behavior of the server / router. 

Reproducible Example:

```javascript
describe('Route does not work after hash changed', function() {
  beforeEach(function() {
    cy.visit('https://examples.sencha.com/coworkee/#login')
    cy.url().should('include', '#login')
    cy.get('button#ext-element-43').click()
  })

  it('Waiting For Route Works *Without* Hash Change (after login)', function() {
    cy.visit('https://examples.sencha.com/coworkee/#people')
    cy.contains('Benjamin Banks')
    cy.url().should('include', '#people')
    cy.window().then(win => {
      cy.server()
      cy.state('server').bindTo(win)
      cy.route('POST', 'https://examples.sencha.com/coworkee/api').as('api')
      win.Ext.ComponentQuery.query('searchfield')[0].setValue('Diane')
      cy.wait('@api')
      cy.contains('Moore')
    })
  })

  it('Waiting For Route Fails *With* Hash Change (after login)', function() {
    cy.visit('https://examples.sencha.com/coworkee/#people')
    cy.contains('Benjamin Banks')
    cy.url().should('include', '#people')
    cy.window().then(win => {
      cy.server()
      cy.state('server').bindTo(win)
      cy.route('POST', 'https://examples.sencha.com/coworkee/api').as('api')
      win.Ext.ComponentQuery.query('mainmenu')[0]
        .getController()
        .redirectTo('people')
      cy.wait('@api')
      cy.url().should('include', '#people')
    })
  })
})
```

Fork of cypress-test-tiny to reproduce: https://github.com/urbicon/cypress-test-tiny/tree/5629-Route-not-working-after-hash-changed

### Versions

Cypress: 3.4.1
Browser: Electron 61
OS: MacOS 10.15.1

**Comment by jennifer-shehane** [comment]
The test above passes when I run it in 3.4.1 or 3.8.1 of Cypress. 

<img width="1354" alt="Screen Shot 2020-01-06 at 12 00 43 PM" src="https://user-images.githubusercontent.com/1271364/71797413-63671780-307c-11ea-9c63-281123c32e4b.png">

I'll be closing this issue as it is no longer reproducible from the example provided. Please provide a reproducible example of the bug and we will consider reopening.

**Comment by jennifer-shehane** [comment]
We have a potential fix available for this starting in [`4.6.0`](https://docs-staging.cypress.io/guides/references/changelog.html#4-6-0) as an [experiment](https://on.cypress.io/experiments).Can you try setting this config option in your `cypress.json` or elsewhere:

```json
{
	"experimentalSourceRewriting": true
}
```

**Comment by sainthkh** [comment]
Closing this issue because the example doesn't work (click to login fails) and there is no response to experimental solution. If the problem is still happening, please open a new issue with a reproducible example.
